src/Orleans.Streaming/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -1,453 +1,497 @@
+    }
 
-        private readonly LeaseBasedQueueBalancerOptions options;
-        private readonly ILeaseProvider leaseProvider;
-        private readonly ITimerRegistry timerRegistry;
-        private readonly AsyncSerialExecutor executor = new AsyncSerialExecutor();
-        private int allQueuesCount;
-        private readonly List<AcquiredQueue> myQueues = new List<AcquiredQueue>();
-        private IDisposable leaseMaintenanceTimer;
-        private IDisposable leaseAquisitionTimer;
-        private RoundRobinSelector<QueueId> queueSelector;
-        private int responsibility;
-        private int leaseOrder;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="LeaseBasedQueueBalancer"/> class.
-        /// </summary>
-        /// <param name="name">The name.</param>
-        /// <param name="options">The options.</param>
-        /// <param name="leaseProvider">The lease provider.</param>
-        /// <param name="timerRegistry">The timer registry.</param>
-        /// <param name="services">The services.</param>
-        /// <param name="loggerFactory">The logger factory.</param>
-        public LeaseBasedQueueBalancer(
-            string name,
-            LeaseBasedQueueBalancerOptions options,
-            ILeaseProvider leaseProvider,
-            ITimerRegistry timerRegistry,
-            IServiceProvider services,
-            ILoggerFactory loggerFactory)
-            : base(services, loggerFactory.CreateLogger($"{typeof(LeaseBasedQueueBalancer).FullName}.{name}"))
+    /// <inheritdoc/>
+    public override async Task Initialize(IStreamQueueMapper queueMapper)
+    {
+        if (Cancellation.IsCancellationRequested)
         {
-            this.options = options;
-            this.leaseProvider = leaseProvider;
-            this.timerRegistry = timerRegistry;
+            throw new InvalidOperationException("Cannot initialize a terminated balancer.");
         }
 
-        /// <summary>
-        /// Creates a new <see cref="LeaseBasedQueueBalancer"/> instance.
-        /// </summary>
-        /// <param name="services">The services.</param>
-        /// <param name="name">The name.</param>
-        /// <returns>The new <see cref="LeaseBasedQueueBalancer"/> instance.</returns>
-        public static IStreamQueueBalancer Create(IServiceProvider services, string name)
+        ArgumentNullException.ThrowIfNull(queueMapper);
+        var allQueues = queueMapper.GetAllQueues().ToList();
+        _allQueuesCount = allQueues.Count;
+
+        // Selector default to round robin selector now, but we can make a further change to make selector configurable if needed. Selector algorithm could 
+        // be affecting queue balancing stabilization time in cluster initializing and auto-scaling
+        _queueSelector = new RoundRobinSelector<QueueId>(allQueues);
+        await base.Initialize(queueMapper);
+        StartMaintenanceTasks();
```

Comment:
Always start the timer methods - these run for the lifetime of the class. We pause/resume/expedite timers by changing their `Period` property.